### PR TITLE
Fix for File Picker Types Array Handling in TinyMce Integration

### PR DESCRIPTION
**Pull request by shoaib0300**
**Problem:**
Previously, passing the `fileBrowserTypes` array to `file_picker_types` caused a warning due to incorrect array-to-string conversion.

**Solution:**
This fix properly encodes `fileBrowserTypes` using `json_encode()`, ensuring `file_picker_types` receives a correctly formatted JSON string. It also sets default values using `explode('|', $field['eval']['rte'], 2) + [null, null]` to handle cases where `$field['eval']['rte']` may not contain both parts.

**Testing:**
- Verified compatibility by testing in various Contao versions.
- Confirmed that `file_picker_types` now receives valid JSON data for expected behavior.

### CODE REVIEW

**Comment by zonky2** [comment]
@stefanheimes LGTM

File: src/EventListener/Mcw/TinyMce.php [hunk 1]
```diff
@@ -63,7 +63,7 @@ public function executeEvent(GetTinyMceStringEvent $event)
         $objTemplate                   = new BackendTemplate('be_' . $file);
         $objTemplate->selector         = 'ctrl_' . $fieldId;
         $objTemplate->type             = $type;
-        $objTemplate->fileBrowserTypes = $fileBrowserTypes;
+        $objTemplate->fileBrowserTypes =  json_encode($fileBrowserTypes);
```
Comment (fritzmg) [inline src/EventListener/Mcw/TinyMce.php]: This will not work in Contao 4.13. See https://github.com/contao/contao/blob/fb078fd3c1e8b6c81f2743ec01736cf7f70deb55/core-bundle/src/Resources/contao/templates/backend/be_tinyMCE.html5#L44

**Review by fritzmg — CHANGES_REQUESTED** [review]

**Comment by shoaib0300** [comment]
Thank you, @fritzmg, for pointing this out. I do have the Contao 4.13 project set up; I will have a look at it this weekend.

File: src/EventListener/Mcw/TinyMce.php (hunk 1 above)
Comment (shoaib0300) [reply src/EventListener/Mcw/TinyMce.php]: @fritzmg  I understand your concern regarding the behavior in Contao 4.13 and the reference to be_tinyMCE.html5. However, after thorough testing, I can confirm that the fix works as expected.
Below are video recording showing the working behavior in Contao 4.13.

If there's any specific scenario or configuration in Contao 4.13 where this might break, I'd be happy to test further and adjust the implementation if needed!

Looking forward to your thoughts. 😊


https://github.com/user-attachments/assets/7ebc46bd-d740-45e1-ba2c-a1744705e677

Comment (fritzmg) [reply src/EventListener/Mcw/TinyMce.php]: What is the source code output for this? Wouldn't this double json encode the `$fileBrowserTypes`? Your video did not show a file picker test.

Comment (shoaib0300) [reply src/EventListener/Mcw/TinyMce.php]: @fritzmg Could you please review my second commit? I made some changes to fix Contao 4.13 and Contao 5. It works fine; I can upload files and the file picker option is also available.

Comment (fritzmg) [reply src/EventListener/Mcw/TinyMce.php]: Hm, you are right, the `file_picker_types` setting does not seem to have any discernible effect 🤔 No matter what you put there.

File: src/Contao/Widgets/MultiColumnWizard.php [hunk 2]
```diff
@@ -1071,16 +1071,21 @@ protected function initializeWidget(&$arrField, $intRow, $strKey, $varValue)
...
-                       ) . '</a>';
+                . '" title="' . StringUtil::specialchars($GLOBALS['TL_LANG']['MSC']['helpWizard'])
+                . '" onclick="Backend.openModalIframe({\'width\':735,\'height\':405,\'title\':\''
+                . StringUtil::specialchars(str_replace("'", "\\'", $label))
+                . '\',\'url\':this.href});return false">'
+                . Image::getHtml(
+                    'about.gif',
+                    $GLOBALS['TL_LANG']['MSC']['helpWizard'],
+                    'style="vertical-align:text-bottom"'
+                ) . '</a>';
```
Comment (fritzmg) [inline src/Contao/Widgets/MultiColumnWizard.php:1088]: What is the reason for this change? The label should always be there.

Comment (shoaib0300) [reply src/Contao/Widgets/MultiColumnWizard.php:1088]: The modification was introduced to address situations where the $arrField['label'][0] was not defined, leading to warnings like Trying to access array offset on value of type null. This could happen in edge cases where the label configuration is either incomplete or missing.

Comment (shoaib0300) [reply src/Contao/Widgets/MultiColumnWizard.php:1088]: There was and error
Trying to access array offset on value of type null

The error indicates that $arrField['label'] is null or not set, so accessing [0] on it results in the warning. To resolve this issue, I check if $arrField['label'] is properly defined and is an array before trying to access its elements.

File: src/EventListener/Mcw/TinyMce.php (hunk 1 above)
Comment (shoaib0300) [reply src/EventListener/Mcw/TinyMce.php]: @fritzmg i don't get your point but for me contao 4.13 I can see file picker and I uploaded files also see the screenshots below

image
image
image

Comment (fritzmg) [reply src/EventListener/Mcw/TinyMce.php]: It turns out that TinyMCE supports both arrays and also space or comma separated strings for that setting - so your current solution works for both Contao 4.13 and 5 👍

Comment (shoaib0300) [reply src/EventListener/Mcw/TinyMce.php]: @fritzmg  Thanks for confirming! Glad the solution works for both versions. 👍

**Comment by shoaib0300** [comment]
@fritzmg 
I noticed that the pull request has been reviewed. I just wanted to check if there’s a timeline for when it will be merged. I want to plan to use the changes and would appreciate it if you could let me know the expected timeframe.

Thanks 😊

**Comment by fritzmg** [comment]
@shoaib0300 I am not a maintainer of this package.

File: src/Contao/Widgets/MultiColumnWizard.php (hunk 2 above)
Comment (fritzmg) [reply src/Contao/Widgets/MultiColumnWizard.php:1088]: But how can this be reproduced?

Comment (shoaib0300) [reply src/Contao/Widgets/MultiColumnWizard.php:1088]: I was testing different PHP versions and various Contao versions. While fixing TinyMCE for multiwizardcolumn, I encountered that issue. I'm trying to reproduce it again, and if I find anything, I will update you.
